test/jdk/jdk/jfr/event/oldobject/TestObjectSize.java

```diff
@@ -100,13 +97,13 @@ public static void main(String[] args) throws Exception {
                     if (objectSize <= 0) {
                         throw new Exception("Object size for " + type.getName() + " is lower or equal to 0");
                     }
-                    if (type.getName().equals(Leak1.class.getName())) {
+                    if (type.getName().equals(Leak1.class.getName()) && sizeLeak1 == -1) {
                         sizeLeak1 = objectSize;
                     }
-                    if (type.getName().equals(Leak2.class.getName())) {
+                    if (type.getName().equals(Leak2.class.getName()) && sizeLeak2 == -1) {
                         sizeLeak2 = objectSize;
                     }
-                    if (type.getName().equals(Leak3.class.getName())) {
+                    if (type.getName().equals(Leak3.class.getName()) && sizeLeak3 == -1) {
```

Comment:
So, this fix about not rewriting `sizeLeakN` variable in subsequent retries as each retry might have only a subset of those variables properly set, correct?

Reply:
Correct.